### ISSUE TITLE
Error with safe navigation operator and assignment

### ISSUE THREAD

**Issue by gkop** [seattlerb/ruby_parser#214]
I'm trying to figure out what's going wrong when I call `ruby_parse` on this input and get `#<Racc::ParseError: bar.rb:1 :: parse error on value ":" (tCOLON)> for bar.rb`:

``` ruby
# in bar.rb
bar(baz: true)

burp = herp&.derp
```

I can make another simple case that results in `#<Racc::ParseError: foo.rb:1 :: parse error on value "&." (error)>`:

``` ruby
# in foo.rb
foo = baz&.to_s
```

Using Ruby 2.3.0 and ruby_parser 3.8.1

**Comment by dsferreira** [comment]
According to this: https://github.com/seattlerb/ruby_parser/commit/a966b2b7cf62567a34292561b7f00606e590a3bc

ruby 2.3.0 should by now be supported.

The issue with the save navigation operator still holds.
Shouldn't this issue be reopen?

There is a workaround though for anyone facing the same issue:

``` ruby
a = (b&.c)
```

**Comment by phiggins** [comment]
I've run into this as well.

RubyParser can parse the strings `foo = bar&.baz`, `bar&.baz;`, `bar&.bar\n`, `foo&.bar.baz;` and `foo&.bar.baz\n`. It cannot parse the strings `foo = bar&.baz;` or `foo = bar&.baz\n`.

```
2.3.0 :001 > RubyParser.new.process "foo = bar&.baz"
 => s(:lasgn, :foo, s(:safe_call, s(:call, nil, :bar), :baz)) 
2.3.0 :002 > RubyParser.new.process "bar&.baz\n"
 => s(:safe_call, s(:call, nil, :bar), :baz) 
2.3.0 :003 > RubyParser.new.process "bar&.baz;"
 => s(:safe_call, s(:call, nil, :bar), :baz) 
2.3.0 :004 > RubyParser.new.process "foo = bar&.baz;"
Racc::ParseError: (string):1 :: parse error on value "&." (error)
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/racc/parser.rb:528:in `on_error'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1125:in `on_error'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/racc/parser.rb:259:in `_racc_do_parse_c'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/racc/parser.rb:259:in `do_parse'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1037:in `block in process'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:33:in `catch'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:33:in `catch'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1025:in `process'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1368:in `block in process'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1366:in `each'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1366:in `process'
	from (irb):4
	from /home/pete/.rvm/rubies/ruby-2.3.0/bin/irb:11:in `<main>'
2.3.0 :005 > RubyParser.new.process "foo = bar&.baz\n"
Racc::ParseError: (string):1 :: parse error on value "&." (error)
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/racc/parser.rb:528:in `on_error'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1125:in `on_error'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/racc/parser.rb:259:in `_racc_do_parse_c'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/racc/parser.rb:259:in `do_parse'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1037:in `block in process'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:33:in `catch'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:33:in `catch'
	from /home/pete/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1025:in `process'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1368:in `block in process'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1366:in `each'
	from /home/pete/.rvm/gems/ruby-2.3.0@debride/gems/ruby_parser-3.8.1/lib/ruby_parser_extras.rb:1366:in `process'
	from (irb):5
	from /home/pete/.rvm/rubies/ruby-2.3.0/bin/irb:11:in `<main>'
2.3.0 :006 > 

# ...

2.3.0 :013 >   RubyParser.new.process "foo = bar&.baz.qux;"
 => s(:lasgn, :foo, s(:call, s(:safe_call, s(:call, nil, :bar), :baz), :qux)) 
2.3.0 :014 > RubyParser.new.process "foo = bar&.baz.qux\n"
 => s(:lasgn, :foo, s(:call, s(:safe_call, s(:call, nil, :bar), :baz), :qux)) 
```

**Comment by presidentbeef** [comment]
@zenspider: fix for this pending. I finally looked at Ruby's parser like I should have done the first time around and I think their approach should work fine.

**Comment by phiggins** [comment]
Looks like this was fixed by #217. Thanks @presidentbeef and @zenspider!

**Comment by zenspider** [comment]
confirmed long fixed. I need to triage more often.  Thank you everyone.
